pkg/jobrunaggregator/jobrunaggregatoranalyzer/analyzer_disruption.go

```diff
@@ -0,0 +1,174 @@
+package jobrunaggregatoranalyzer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v2"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+func (o *JobRunAggregatorAnalyzerOptions) CalculateDisruptionTestSuite(ctx context.Context, jobName string, finishedJobsToAggregate []jobrunaggregatorapi.JobRunInfo) (*junit.TestSuite, error) {
+	disruptionJunitSuite := &junit.TestSuite{
+		Name:      "BackendDisruption",
+		TestCases: []*junit.TestCase{},
+	}
+	collectedDataTestCase := &junit.TestCase{
+		Name: "should collect disruption data",
+	}
+	disruptionJunitSuite.TestCases = append(disruptionJunitSuite.TestCases, collectedDataTestCase)
+
+	jobRunIDToBackendNameToAvailabilityResult, err := getDisruptionByJobRunID(ctx, finishedJobsToAggregate)
+	if jobRunIDToBackendNameToAvailabilityResult != nil {
+		rawDataBytes, err := yaml.Marshal(jobRunIDToBackendNameToAvailabilityResult)
+		if err != nil {
+			collectedDataTestCase.SystemOut = string(rawDataBytes)
+		}
+	}
+	switch {
+	case len(jobRunIDToBackendNameToAvailabilityResult) < 3 && err != nil:
+		return nil, err
+	case len(jobRunIDToBackendNameToAvailabilityResult) < 3 && err == nil:
+		collectedDataTestCase.FailureOutput = &junit.FailureOutput{
+			Message: "not enough data to aggregate",
+			Output:  collectedDataTestCase.SystemOut,
+		}
+		disruptionJunitSuite.NumFailed++
+		return disruptionJunitSuite, nil
+
+	default:
+		// ignore the errors if we have at least three results
+		fmt.Fprintf(os.Stderr, "Could not fetch backend disruption data for all runs %v\n", err)
+	}
+
+	allBackends := getAllDisruptionBackendNames(jobRunIDToBackendNameToAvailabilityResult)
+	for _, backendName := range allBackends.List() {
+		jobRunIDToAvailabilityResultForBackend := getDisruptionForBackend(jobRunIDToBackendNameToAvailabilityResult, backendName)
+		historicalStats, failed, message, err := o.passFailCalculator.CheckDisruption(ctx, jobRunIDToAvailabilityResultForBackend, backendName)
```

Comment:
This shouldn't be called CheckDisruption anymore

Reply:
Maybe I'm wrong, I'm a little confused what's going on here compared to the for loop below. I need to look with fresh eyes tomorrow

Reply:
> Maybe I'm wrong, I'm a little confused what's going on here compared to the for loop below. I need to look with fresh eyes tomorrow

I wanted to keep the diff small. I'm open to renaming this or even moving it if you like.  The fail, message, and err from this method are honored.  the historicalstats allow synthetic junitest creation.